Pods/AppDevSettings/AppDevSettings/Classes/SettingsViewController.swift

```diff
@@ -0,0 +1,142 @@
+
+public class SettingsViewController: UITableViewController {
+
+    private var config: SettingsConfig!
+
+    private enum CellIdentifiers {
+        public static let booleanCell = "booleanCell"
+        public static let buttonCell = "buttonCell"
+        public static let inputCell = "inputCell"
+    }
+
+    public init(config: SettingsConfig) {
+
+        super.init(style: .grouped)
+
+        self.config = config
+
+        let headerImageBuffer: CGFloat = 20
+
+        if let image = config.headerImage, let h = config.headerHeight {
+
+            let headerView = UIView(frame: CGRect(x: 0, y: 0, width: view.frame.width, height: h + 2*headerImageBuffer))
+
+            let imageView = UIImageView(image: image)
+            imageView.contentMode = .scaleAspectFit
+            headerView.addSubview(imageView)
+
+            imageView.snp.makeConstraints { make in
+                make.leading.trailing.equalToSuperview()
+                make.top.bottom.equalToSuperview().inset(headerImageBuffer)
+            }
+
+            tableView.tableHeaderView = headerView
+
+        } else if let view = config.headerView, let h = config.headerHeight {
+
+            let headerView = UIView(frame: CGRect(x: 0, y: 0, width: view.frame.width, height: h))
+            headerView.addSubview(view)
+
+            view.snp.makeConstraints { make in
+                make.edges.equalToSuperview()
+            }
+
+            tableView.tableHeaderView = headerView
+
+        }
+
+        self.tableView.register(BooleanTableViewCell.self, forCellReuseIdentifier: CellIdentifiers.booleanCell)
+        self.tableView.register(ButtonTableViewCell.self, forCellReuseIdentifier: CellIdentifiers.buttonCell)
+        self.tableView.register(InputTableViewCell.self, forCellReuseIdentifier: CellIdentifiers.inputCell)
+
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+
+        title = "Settings"
+
+        navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(didPressDone))
+    }
+
+    override public func numberOfSections(in tableView: UITableView) -> Int {
+        return config.sectionsOfItems.count
+    }
+
+    override public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return config.sectionsOfItems[section].count
+    }
+
+    override public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+
+        let item = config.sectionsOfItems[indexPath.section][indexPath.row]
+
+        switch item {
+        case .bool(let title, let key, let compl):
+            let cell = tableView.dequeueReusableCell(withIdentifier: CellIdentifiers.booleanCell) as! BooleanTableViewCell
+            cell.title = title
+            cell.key = key
+            cell.completion = compl
```

Comment:
Can we rename compl to completion so it's consistent with line 95 and the full names for title/key